### PR TITLE
refactor: use dataclass for weight axis defaults table

### PR DESCRIPTION
**Pull request by pkmandke**
Use a `dataclass` instead of simple dict for the axis defaults table.

### CODE REVIEW

File: src/coreai_opt/quantization/_axis_defaults.py [hunk 1]
```diff
@@ -35,43 +36,45 @@
...
+    nn.Conv3d: _WeightAxisSpec(0, 1, (torch.ops.aten.conv3d.default,)),
+    nn.ConvTranspose1d: _WeightAxisSpec(1, 0, (torch.ops.aten.conv_transpose1d.default,)),
+    nn.ConvTranspose2d: _WeightAxisSpec(1, 0, (torch.ops.aten.conv_transpose2d.input,)),
+    nn.ConvTranspose3d: _WeightAxisSpec(1, 0, (torch.ops.aten.conv_transpose3d.input,)),
+    nn.Linear: _WeightAxisSpec(0, 1, (torch.ops.aten.linear.default,)),
+    nn.Embedding: _WeightAxisSpec(0, 1, (torch.ops.aten.embedding.default,)),
 }
 
-# Maps aten OpOverload -> nn.Module type for graph-mode op to module type resolution.
 _ATEN_OP_TO_MODULE_TYPE: dict[torch._ops.OpOverload, type[nn.Module]] = {
```
Comment (crowbat) [inline src/coreai_opt/quantization/_axis_defaults.py]: This mapping in particular doesn't seem specific to axis defaults but more general to torch. Can we move it to `torch_utils.py` and have it define the mapping explicitly instead of pulling from `_WEIGHT_AXIS_SPECS` (which doesn't need to own the mapping)

File: tests/quantization/test_axis_defaults.py [hunk 2]
```diff
@@ -184,9 +183,9 @@ def test_axis_none_resolved(
         prepared = Quantizer(model, config).prepare((make_input(),))
 
         if isinstance(granularity, PerChannelGranularity):
-            expected_axis = _PER_CHANNEL_WEIGHT_AXIS_DEFAULTS[module_type]
+            expected_axis = _WEIGHT_AXIS_SPECS[module_type].per_channel_axis
```
Comment (crowbat) [inline tests/quantization/test_axis_defaults.py:187]: Can we also test the `default_axis_for` function here?

File: src/coreai_opt/quantization/_axis_defaults.py (hunk 1 above)
Comment (pkmandke) [reply src/coreai_opt/quantization/_axis_defaults.py]: makes sense!

My intention here was that when we add a new entry for defaults, we only change it in one place. But with a separate table for the aten to module mapping, it will be 2 places.. I think that's fine since we get the mapping as a util of its own, separate from the defaults.

Comment (crowbat) [reply src/coreai_opt/quantization/_axis_defaults.py]: I see, yeah I feel like the table is generic enough that other things may also make use of it, feels better to separate out the responsibilities in this case.

**Review by crowbat — APPROVED** [review]
